doc/tokens.md

```diff
@@ -1,148 +1,251 @@
+JWT is currently checked in 2 places:
+
+- when a user connects to Prosody through BOSH or websocket. The token value is passed as the 'token' query parameter of 
+  the BOSH or websocket URL. User uses XMPP anonymous authentication method.
+- when a MUC room is being created/joined Prosody compares the 'room' claim with the actual name of the room. In
+  addition, the 'sub' claim is compare to either the tenant (for TENANT/ROOM URLs) or the base domain (for /ROOM URLs).
+  This prevents a stolen token being abused by unauthorized users to allocate new conference rooms in the system. Admin
+  users are not required to provide a valid token which is used for example by Jicofo.
 
 ### Lib-jitsi-meet options
 
-When JWT authentication is used with *lib-jitsi-meet* the token is passed to the *JitsiConference* constructor:
+When JWT authentication is used with _lib-jitsi-meet_ the token is passed to the _JitsiConference_ constructor:
 
-```
-var token = {token is provided by your application possibly after some authentication}
+```javascript
+var token = {token_is_provided_by_your_application_possibly_after_some_authentication}
 
-JitsiMeetJS.init(initOptions).then(function(){
+JitsiMeetJS.init(initOptions).then(function () {
     connection = new JitsiMeetJS.JitsiConnection(APP_ID, token, options);
-    ...
+    // ...
     connection.connect();
 });
-
 ```
 
 ### Jitsi-meet options
 
 In order to start a jitsi-meet conference with a token you need to specify the token as an URL param:
-```
-https://example.com/angrywhalesgrowhigh?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ
-```
-At the current level of integration every user that joins the conference has to provide the token and not just the one who
-creates the room. It should be possible to change that by using a second anonymous domain, but that hasn't been tested
-yet.
 
+`
+https://example.com/angrywhalesgrowhigh?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ
+`
 
+At the current level of integration every user that joins the conference has to provide the token and not just the one
+who creates the room. It should be possible to change that by using a second anonymous domain, but that hasn't been
+tested yet.
 
 ### Installing the token plugin
 
-Token authentication can be integrated automatically using a Debian package install. Once you have jitsi-meet installed
-just install 'jitsi-meet-tokens' on top of it. In order to have it configured automatically at least version 779 of
+Token authentication can be integrated automatically using a Debian package. Once you have `jitsi-meet` installed
+just install `jitsi-meet-tokens` on top of it. In order to have it configured automatically at least version 779 of
```

Comment:
I believe this package will be going away soon, right @damencho ?

Reply:
Well we have talked about it, but nothing is planned ... I think we need to figure out and move it forward when doing the authentication refactor.